src/org/omegat/core/team2/impl/GITRemoteRepository2.java

```diff
@@ -382,6 +380,34 @@ public static boolean deleteDirectory(File path) {
         return (path.delete());
     }
 
+    /**
+     * Retrieve default branch name from remote origin HEAD.
+     * @param repositoryUrl Remote repository URL
+     * @return default branch name, ordinary "main"(recent popular) or "master"(old default)
+     */
+    public static String getDefaultBranchName(final String repositoryUrl) {
+        try {
+            Map<String, Ref> gitMap = Git.lsRemoteRepository().setRemote(repositoryUrl).callAsMap();
+            Ref head = gitMap.get("HEAD");
+            if (head.isSymbolic()) {
+                return getBranchFromRef(head.getTarget().getName());
+            } else {
+                for (String refname : gitMap.keySet()) {
+                    if (refname.startsWith("refs/heads/")
+                            && head.getObjectId().equals(gitMap.get(refname).getObjectId())) {
+                        return getBranchFromRef(refname);
+                    }
+                }
+            }
+        } catch (NullPointerException | GitAPIException ignore) {
+        }
+        return "master";
+    }
+
+    private static String getBranchFromRef(String refName) {
+        return refName.substring(refName.lastIndexOf('/') + 1);
```

Comment:
I don't think this is right in the general case.

Branch names can (and often do) contain `/`. Maybe you just need to trim the prefix `refs/heads/`?